hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/conf/Configuration.java

```diff
@@ -2483,6 +2483,38 @@ public char[] getPasswordFromCredentialProviders(String name)
     return pass;
   }
 
+  /**
+   * Try and resolve the provided element name as a credential provider
+   * alias from the given provider path.
+   * @param name alias of the provisioned credential
+   * @param providerURI The URI of the provider path
+   * @return password or null if not found
+   * @throws IOException when error in fetching password
+   */
+  public char[] getPasswordFromCredentialProvider(String name, URI providerUri)
+      throws IOException {
+    try {
+      CredentialProvider provider = CredentialProviderFactory.getProvider(this,
+          providerUri);
+      if (provider != null) {
+        try {
+          CredentialEntry entry = getCredentialEntry(provider, name);
+          if (entry != null) {
+            return entry.getCredential();
+          }
+        } catch (IOException ioe) {
+          String msg = String.format(
+              "Can't get key %s from key provider of type: %s.", name,
+              provider.getClass().getName());
+          throw NetUtils.wrapWithMessage(ioe, msg);
+        }
+      }
+    } catch (IOException ioe) {
```

Comment:
this will ignore the work done on L2509; really wrapWithMessage() should be invoked for all, and exactly once.